### ISSUE TITLE
Does coiniumserv deal with Masternodes correct? If so what is required?

### ISSUE THREAD

**Issue by Virtuworks** [bonesoul/CoiniumServ#830]

**Comment by vagabondan** [comment]
BigInteger constructor from byte[] should be fixed because it creates negative integers which is improper while comparing with target. This should fix ERRORS like:
high-hash, proof of work failed (code 16)

**Comment by vagabondan** [comment]
For LTC bug was found:
AcceptBlock FAILED: bad-txnmrklroot, hashMerkleRoot mismatch (code 16)
I believe that is because of segwit adoption in LTC and not only RPC getblocktemplate call should be  fixed, but transaction hashes calculation is also affected.

**Comment by Virtuworks** [comment]
@vagabondan yes I agree with you, I found another stratum that looks to have a complete implementation: 

https://github.com/slush0/stratum-mining/pull/16

I think this is the type of change that needs to be implemented to Bitcoin and Litecoin 

@Vagabondan would be great to establish some sort of ongoing conversation I'm available on Skype.

**Comment by vagabondan** [comment]
@Virtuworks yes, and the description of changes needed is here:
https://github.com/bitcoin/bips/blob/master/bip-0145.mediawiki

Skype is not very convenient, but possible. Could you contact me through telegram? Vagabondan is my nick there.

**Comment by ArthurGrizzle** [comment]
Good morning people
